insights/parsers/nginx_conf.py

```diff
@@ -0,0 +1,338 @@
+    #      semicolon situations
+    # Addresses https://github.com/fatiherikli/nginxparser/issues/19
+    map_pattern = Regex(r'".*"') | Regex(r"'.*'") | nonspace
+    map_entry = space + map_pattern + space + value + space + semicolon
+    map_block = Group(
+        Group(map_statement).leaveWhitespace() +
+        left_bracket +
+        Group(ZeroOrMore(Group(comment | map_entry)) + space).leaveWhitespace() +
+        right_bracket)
+
+    block = Forward()
+
+    # key could for instance be "server" or "http", or "location" (in which case
+    # location_statement needs to have a non-empty location)
+
+    block_begin = (Group(space + key + location_statement) ^
+                   Group(if_statement) ^
+                   Group(charset_map_statement)).leaveWhitespace()
+
+    block_innards = Group(ZeroOrMore(Group(comment | assignment) | block | map_block) + space).leaveWhitespace()
+
+    block << Group(block_begin + left_bracket + block_innards + right_bracket)
+
+    script = OneOrMore(Group(comment | assignment) ^ block ^ map_block) + space + stringEnd
+    return script.parseWithTabs().leaveWhitespace()
+
+
+def spacey(x):
+    return (isinstance(x, str) and x.isspace()) or x == ''
+
+
+class UnspacedList(list):
+    """After handling by create_parser(), there is white space existing in the list. Use this class to wrap a list
+     [of lists], making any whitespace entries magically invisible"""
+
+    def __init__(self, list_source):
+        # ensure our argument is not a generator, and duplicate any sublists
+        self.spaced = copy.deepcopy(list(list_source))
+        self.dirty = False
+
+        # Turn self into a version of the source list that has spaces removed
+        # and all sub-lists also UnspacedList()ed
+        list.__init__(self, list_source)
+        for i, entry in reversed(list(enumerate(self))):
+            if isinstance(entry, list):
+                sublist = UnspacedList(entry)
+                list.__setitem__(self, i, sublist)
+                self.spaced[i] = sublist.spaced
+            elif spacey(entry):
+                # don't delete comments
+                if "#" not in self[:i]:
+                    list.__delitem__(self, i)
+
+
+def _handle_key_value(t_dict, key, value):
+    """
+    Function to handle key has multi value, and return the values as list.
+    """
+    if key in t_dict:
+        val = t_dict[key]
+        if isinstance(val, dict):
+            val = [val]
+        val.append(value)
+        return val
+    return value
+
+
+def depth(l):
+    """
+    Function to count the depth of a list
+    """
+    if isinstance(l, list):
+        return 1 + max(depth(item) for item in l)
+    else:
+        return 0
+
+
+def convert_nginx_list_to_dict(li):
+    """
+    After parsed by create_parser(), the result is like following, it is better to convert to dict for convenience.
+    Example: [['user', 'root'], ['worker_processes', '5'], ['error_log', 'logs/error.log'], ['pid',
```

Comment:
You can probably afford to trim this down a bit to make your example more obvious.